src/utils/preflight.py

```
@@ -0,0 +1,252 @@
+
+    REQUIRED_CREDENTIALS = [
+        ("github_token", "GitHub Token", "Required for LLM and embeddings"),
+        ("surrealdb_url", "SurrealDB URL", "Database connection"),
+        ("surrealdb_user", "SurrealDB Username", "Database authentication"),
+        ("surrealdb_pass", "SurrealDB Password", "Database authentication"),
+    ]
+
+    OPTIONAL_CREDENTIALS = [
+        ("google_api_key", "Google API Key", "Required for web search"),
+        ("google_cse_id", "Google CSE ID", "Required for web search"),
+    ]
```

Ruff (0.15.5)

[warning] Mutable default value for class attribute

(RUF012)